src/NBomber/Domain/Domain.fs

```diff
@@ -226,6 +228,53 @@ module internal Scenario =
         if errors.Length > 0 then Error <| FlowErrors(errors)
         else Ok <| scenario
 
+module Assertions =
+
+    let apply (scenarioName: string, flows: AssertionStats[], assertions: Assertion[]) =         
+       assertions 
+       |> Array.mapi (fun i assertion -> applyAssertion(scenarioName, flows, i+1, assertion))
+       |> printAssertionResults
+
+    let private applyAssertion(scenarioName: string, flows: AssertionStats[], i: int, assertion: Assertion) =
+       match assertion with
+           | Scenario (func) ->
+                let applied = flows
+                            |> Array.groupBy (fun f -> f.FlowName)
+                            |> Array.map (fun (_, steps) -> applyForSteps(steps, func))
+                            |> Array.exists (fun (result) -> match result with | Some(x) -> x | None -> true)
+                            |> Some
+
+                createAssertionResult(applied, "Scenario", scenarioName, i)
+
+           | TestFlow (flowName, func) ->
+                let steps = flows |> Array.where (fun f -> f.FlowName = flowName)
+                let applied = applyForSteps(steps, func)
+
+                createAssertionResult(applied, "Test Flow", flowName, i)
+
+           | Step (flowName, stepName, func) -> 
+               let steps = flows |> Array.filter (fun x -> x.FlowName = flowName && x.StepName = stepName)
+               let applied = applyForSteps(steps, func)
+
+               createAssertionResult(applied, "Step", stepName, i)
+
+    let private applyForSteps (steps: AssertionStats[], assertion: AssertionFunc) =
+        let applied = steps |> Array.map(assertion.Invoke)
+        match applied with | [||] -> None | stepResults -> stepResults |> Array.exists(id) |> Some
```

Comment:
wrap it to local nested function.  from the code it's quite hard to guess what is actually doing